### PR TITLE
Adding Friktion SDK interface

### PR DESCRIPTION
**Pull request by mdantonio**
Now that the SDK interface is merged we can start working on extending it to Friktion

To ease the work I prepared a base setup: I do not expect that it is working but can be a starting point to avoid you starting from scratch

Please note that here I extended get_otoken_details with offer_id and seller parameters and get_offer_details with seller parameter by adding it to both the abstract config and to the Friktion config. Other venues will ignore the new parameters (consumed via kwargs).
Our side, just updated the drafted PR to send the new parameters

### CODE REVIEW

**Review by pdonorio — COMMENTED** [review]
all clear here, let's see what Cassios can say

File: friktion/friktion/config.py
Comment (cassios) [inline friktion/friktion/config.py:115]: signatures on solana are expressed as a single string.
please refer to this open PR https://github.com/tradeparadigm/mono/pull/4768

File: friktion/friktion/config.py
Comment (cassios) [inline friktion/friktion/config.py:141]: verify_allowance interface is being changed. Please refer to https://github.com/tradeparadigm/mono/pull/4768/files#diff-64af62cf35a1f832f00d50df2aea25a565b7d2e549ca2f0e6a5f64a73741d26aR43

**Review by cassios — COMMENTED** [review]
Nice. Just a few comments about changes in progress

File: friktion/friktion/config.py
Comment (mdantonio) [reply friktion/friktion/config.py:141]: Feel free to adapt the implementation with the correct version, you have my complete support for the integration and to keep both the SDK interface and the internal changes in sync (for example if new parameters have to be added). When we will be ready here, we can move on the internal part to have a look together to the integration step

Comment (cassios) [reply friktion/friktion/config.py:141]: ok. I will wait to get the changes reviewed and merged to avoid re-work

Comment (mdantonio) [reply friktion/friktion/config.py:141]: Sure! 💯

**Comment by mdantonio** [comment]
Rebased to include fixes from https://github.com/tradeparadigm/sdks/pull/54,
now we have all tests passing here

@cassios if we have some time here, we could try to complete this interface

**Comment by cassios** [comment]
Sorry, I missed that. I will get back to it next week. I just need to unblock rfq creation for DRFQv2 first

**Review by cassios — APPROVED** [review]
lgtm, just a small comment

great work. Thanks 🎖️

File: friktion/friktion/swap.py
Comment (mdantonio) [reply friktion/friktion/swap.py:126]: Ok!

Just a note:
Due to this issue, we had to [annotate strikePrice as a float](https://github.com/tradeparadigm/sdks/blob/6e04683eeb17b24fa7ff71f5baaf2ae8f7a6b001/sdk_commons/sdk_commons/config.py#L29).
If we can fix here and we want to annotate it as a Decimal,
then we have to change the interface accordingly

File: friktion/friktion/config.py
Comment (mdantonio) [reply friktion/friktion/config.py:100]: Not really. We need to cast it to str only because minPrice is currently [annotated to be a string](https://github.com/tradeparadigm/sdks/blob/6e04683eeb17b24fa7ff71f5baaf2ae8f7a6b001/sdk_commons/sdk_commons/config.py#L13). We can easily change it, but which type should be? (also considering the other venues).
int? float? Decimal?

Comment (cassios) [reply friktion/friktion/config.py:100]: Oh got it. It is not a Decimal at this point as we still have to apply the resolution. So I think we can leave that to adjust as the broader discussion about value resolution goes

File: friktion/friktion/swap.py
Comment (cassios) [reply friktion/friktion/swap.py:126]: Got it. To fix that we could cast the float number to string and then create the Decimal object from that
Screen Shot 2022-08-29 at 14 58 09

Comment (mdantonio) [reply friktion/friktion/swap.py]: Changed, now strikePrice is annotated as a Decimal

**Pull request merged**
